AGameBase/AGameBase.cpp

```diff
@@ -10,98 +10,41 @@ AGameBase::AGameBase() : ABaseEntity() {
 void AGameBase::Tick(float deltaTime) {
     Super::Tick(deltaTime);
 
-    // check for round restart
-    // also check for initialization of all entities (efficient place to do it)
-    if (g_pGlobals->curtime > m_tNextRoundRestart) {
-        // executing round restart statement
-        if (!m_bHasInitializedAllEntities && AllEntitiesReady()) {
-            // initializing all entities
-            InitializeAllEntities();
-        } else if (!AllEntitiesReady()) {
-            //?! This should never happen!
-            ADebug::Assert((bool)(s_iReadyEntityCount == s_iEntityCount),
-                           "s_iReadyEntityCount == s_iEntityCount");
-        } else {
-            // restarting round
-            RestartRound();
-        }
-        m_tNextRoundRestart = FLT_MAX;
+    // by the time Tick is called, all BeginPlay()s have been safely and successfully
+    // called
+    // IBaseEntity::s_bAreBaseEntitiesReady = true;
+    if (!IBaseEntity::s_bHasAlreadyInitializedBaseEntities) {
+        InitializeBaseEntities();
```

Comment:
No need for round restarts and round tracking since we have to remove this all anyways if we want to transition VRBase to a plugin